File: v2-code-samples/PHP/getty_connect_api_searchforimages_p2_paginate.php
<?php

// search string, let's look up "tree"
$searchPhrase = "tree";

// build array to query api for images
$searchImagesArray = array (
	"RequestHeader" => array (
		"Token" => $token // Token received from a CreateSession/RenewSession API call
	),
	"SearchForImages2RequestBody" => array (
 		"Query" => array (
			"SearchPhrase" => $searchPhrase
 		),
 		"ResultOptions" => array (
			"IncludeKeywords" => "false",
 			"ItemCount" => 25, // return 25 items
 			"ItemStartNumber" => 26 // 1-based int, start on the 2nd page
 		)
	)
);

// encode to json
$json = json_encode($searchImagesArray);

$endpoint = "http://connect.gettyimages.com/v1/search/SearchForImages";

// create client and set json data and datatype
$httpClient = new Zend_Http_Client($endpoint);
$httpClient->setRawData($json, 'application/json');
$httpClient->setMethod(Zend_Http_Client::POST); // all getty api requests are POST

// returns Zend_Http_Response
$response = $httpClient->request();

$body = null;

// evaluate for success response
if ($response->isSuccessful()) {
	$body = json_decode($response->getBody()); // returns stdObject
} else {
	// report error
}

// retrieves the image array of stdObjects
$images = $body->SearchForImagesResult->Images;

// iterate through the array of image stdObjects
foreach ($images as $image) {

	// get the image ID
	$imageId = $image->ImageId;

	// get caption of image
	$caption = $image->Caption;

	// get url of the image comp
	$urlComp = $image->UrlComp;

	// get the url of the image thumbnail
	$urlThumb = $image->UrlThumb;

	// print the values for sake of example
	echo "Caption: " . $caption . PHP_EOL;
	echo "Url Comp: " . $urlComp . PHP_EOL;
	echo "Url Thumb: " . $urlThumb . PHP_EOL;
}

?>

File: v2/code-samples/Perl/getty_connect_api_searchforimages_p2_paginate.pl
#!/usr/bin/perl -w
# $Rev: 1 $
# $Application: Getty Connect $
# $Date: 2012-02-21 $
#
# Requires Json
#

use strict;
use warnings;
use JSON;
use CGI;
use LWP::UserAgent;

#set params
my $token = "token";  
my $searchPhrase = "tree";   
my $url = "http://connect.gettyimages.com/v1/search/SearchForImages";
my $itemCount = "25" ; 
my $filters = {"ImageFamily" => "creative"};

# create array of data for request
my $json_data = {
		"RequestHeader" => { 
			"Token" => $token,
			"CoordinationId" => "" 
		},
		"SearchForImages2RequestBody" => {
			"Query"         => { 
				"SearchPhrase" => $searchPhrase 
			},
			"Filter"        => { 
				"ImageFamily"  => $filters->{ImageFamily}
			},
			"ResultOptions" => {
				"IncludeKeywords" => "false",
				"ItemCount"       => $itemCount,
				"ItemStartNumber" => 25
			}
		}
	};


#encode data to json
$json_data = encode_json $json_data;

# post set post request
my $req = HTTP::Request->new( POST => $url );
$req->content_type('application/json');
$req->content($json_data);

# send request
my $ua   = LWP::UserAgent->new;
my $response = $ua->request($req);

#decode data to json
$response  = decode_json $response->content;

#print response
print( $response. "\n" );









